thirdeye-ui/src/app/components/cron-editor/cron-editor.component.tsx

```diff
@@ -0,0 +1,375 @@
+/**
+ * Copyright 2022 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import {
+    Box,
+    FormControl,
+    FormControlLabel,
+    FormHelperText,
+    Grid,
+    MenuItem,
```

Comment:
This component should solely be focused on converting the user friendly input into a cron string. It should also attempt to parse a passed cron string to see if it's simple enough to set the initial values for the hour, minute, and repeat every